bin/sidecli.re

```diff
@@ -0,0 +1,243 @@
+open Helpers;
+open Node;
+open State;
+open Protocol;
+open Cmdliner;
+
+Printexc.record_backtrace(true);
+
+// Helpers
+
+let read_file = file => {
+  let.await ic = Lwt_io.open_file(~mode=Input, file);
+  let.await lines = Lwt_io.read_lines(ic) |> Lwt_stream.to_list;
+  let.await () = Lwt_io.close(ic);
+  await(lines |> String.concat("\n"));
+};
+let read_file = file => read_file(file) |> Lwt_main.run;
+let read_identity_file = file => {
+  let file_buffer = read_file(file);
+  let json = Yojson.Safe.from_string(file_buffer);
+  identity_of_yojson(json) |> Result.get_ok;
+};
+
+// Todo from Andre: we may have to do peer discovery?
+// I don't know anything about that, except for having played with hyperswarm.
+let validators = () => [
+  read_identity_file("0/identity.json"),
+  // read_identity_file("1/identity.json"),
+  read_identity_file("2/identity.json"),
+  // read_identity_file("identity_3.json"),
+];
+
+let make_filename_from_address = wallet_addr_str => {
+  String.concat("", [wallet_addr_str, ".tzsidewallet"]);
+};
+
+module Serializable = {
+  [@deriving yojson]
+  type wallet_file = {
+    address: Wallet.t,
+    priv_key: Address.key,
+  };
+};
+
+let exits =
+  Term.default_exits
+  @ Term.[exit_info(1, ~doc="expected failure (might not be a bug)")];
+
+// Commands
+// ========
+
+// Create wallet
+
+let info_create_wallet = {
+  let doc = "Creates a wallet file. The wallet file's filename is its address. The wallet file contains the private key corresponding to that address.";
+  let man = [
+    `S(Manpage.s_bugs),
+    `P("Email bug reports to <contact@marigold.dev>."),
+  ];
+  Term.info("create-wallet", ~version="%‌%VERSION%%", ~doc, ~exits, ~man);
+};
+
+let create_wallet = () => {
+  let (key, wallet) = Wallet.make_wallet();
+
+  let swallet: Serializable.wallet_file = {priv_key: key, address: wallet};
+  let to_yojson = Serializable.wallet_file_to_yojson;
+  let wallet_json = swallet |> to_yojson;
+
+  let wallet_addr_str = Wallet.address_to_string(wallet);
+
+  let filename = make_filename_from_address(wallet_addr_str);
+
+  Yojson.Safe.to_file(filename, wallet_json);
```

Comment:
We need to handle exceptions here, or use something from Lwt, probably wrap it and return `Lwt.return(`Error(something))`